src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -82,35 +127,59 @@ def validate_functions(functions):
         raise ChatAPIInvalidFunctions(message=f"functions cannot be an empty list. {common_tsg}")
     else:
         for i, function in enumerate(functions):
-            # validate if the function is a dict
-            if not isinstance(function, dict):
-                raise ChatAPIInvalidFunctions(message=f"function {i} '{function}' is not a dict. {common_tsg}")
-            # validate if has required keys
-            for key in ["name", "parameters"]:
-                if key not in function.keys():
-                    raise ChatAPIInvalidFunctions(
-                        message=f"function {i} '{function}' does not have '{key}' property. {common_tsg}")
-            # validate if the parameters is a dict
-            if not isinstance(function["parameters"], dict):
-                raise ChatAPIInvalidFunctions(
-                    message=f"function {i} '{function['name']}' parameters '{function['parameters']}' "
-                            f"should be described as a JSON Schema object. {common_tsg}")
-            # validate if the parameters has required keys
-            for key in ["type", "properties"]:
-                if key not in function["parameters"].keys():
-                    raise ChatAPIInvalidFunctions(
-                        message=f"function {i} '{function['name']}' parameters '{function['parameters']}' "
-                                f"does not have '{key}' property. {common_tsg}")
-            # validate if the parameters type is object
-            if function["parameters"]["type"] != "object":
-                raise ChatAPIInvalidFunctions(
-                    message=f"function {i} '{function['name']}' parameters 'type' "
-                            f"should be 'object'. {common_tsg}")
-            # validate if the parameters properties is a dict
-            if not isinstance(function["parameters"]["properties"], dict):
-                raise ChatAPIInvalidFunctions(
-                    message=f"function {i} '{function['name']}' parameters 'properties' "
-                            f"should be described as a JSON Schema object. {common_tsg}")
+            validate_function(common_tsg, i, function, ChatAPIInvalidFunctions)
+
+
+def validate_tools(tools):
+    """
+        tools = [
+      {
+        "type": "function",
+        "function": {
+          "name": "get_current_weather",
+          "description": "Get the current weather in a given location",
+          "parameters": {
+            "type": "object",
+            "properties": {
+              "location": {
+                "type": "string",
+                "description": "The city and state, e.g. San Francisco, CA",
+              },
+              "unit": {"type": "string", "enum": ["celsius", "fahrenheit"]},
+            },
+            "required": ["location"],
+          },
+        }
+      }
+    ]
+    """
+    tool_example = json.dumps(
+        {
```

Comment:
should have use `type` and `function` to wrap the example?

Reply:
fixed